app/controllers/spree/admin/images_controller.rb

```diff
@@ -50,19 +50,20 @@ def update
         @url_filters = ::ProductFilters.new.extract(request.query_parameters)
         set_viewable
 
-        if @object.update(permitted_resource_params)
+        if @object.update!(permitted_resource_params)
           flash[:success] = flash_message_for(@object, :successfully_updated)
 
           respond_with do |format|
             format.html { redirect_to location_after_save }
             format.turbo_stream
           end
-        else
-          respond_with(@object)
         end
-      rescue ActiveStorage::IntegrityError
-        @object.errors.add :attachment, :integrity_error
-        respond_with(@object)
```

Comment:
Any reason to remove this error handler? I guess we didn't have specs for it.
`±` I'm sure it's there for a good reason so I think we should keep it. I don't know how to replicate it, but I imagine some corrupt files could trigger it.

I would suggest moving the `respond_with` block to a new method, so it can be shared for the two cases. (eg `respond_with_error`)

Reply:
I have refactored to a new `respond_with_error` method.

Reply:
Thanks for looking into the error.  I remember now that I added this check on [this PR](https://github.com/openfoodfoundation/openfoodnetwork/pull/12048#issuecomment-1909206891). 
It has also been tracked here: https://github.com/openfoodfoundation/openfoodnetwork/issues/9302

So it seems that it only happens when trying to view existing images, not when uploading images.
But.. on the `update` action we _do_ also view images; in the `turbo_stream` success format.
So I think if the image integrity has broken, I think we still want to catch the error and let the user know. Because the only way to fix it is by uploading the image again.

However, this is only theory until we can replicate the error. 🤔 If we can't replicate it, then maybe we shouldn't try to catch it. Ok let's leave this out.